### PR TITLE
[test] Use fetch over XMLHttpRequest for browser test reporting. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Inspired by #22015

### CODE REVIEW

File: test/browser/async_returnvalue.cpp [hunk 1]
```diff
@@ -33,18 +33,14 @@ int main() {
...
         Module.reported = true;
         console.log("reporting success");
         // manually REPORT_RESULT; we shouldn't call back into native code at this point
-        var xhr = new XMLHttpRequest();
-        xhr.open("GET", "http://localhost:8888/report_result?0");
-        xhr.onload = xhr.onerror = function() {
-          window.close();
-        };
-        xhr.send();
+        await fetch"http://localhost:8888/report_result?0");
```
Comment (dschuff) [inline test/browser/async_returnvalue.cpp:42]:
```suggestion
        await fetch("http://localhost:8888/report_result?0");
```

File: test/browser/cwrap_early.js [hunk 2]
```diff
@@ -5,13 +5,11 @@ Module['preRun'] = () => {
...
-  Module['onRuntimeInitialized'] = function() {
+  Module['onRuntimeInitialized'] = async () => {
     console.log('onRuntimeInitialized');
     if (wrappedAdd(5, 6) != 11) throw '5 + 6 should be 11';
     // report success
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://localhost:8888/report_result?0', true);
-    xhr.send();
-    setTimeout(function() { window.close() }, 1000);
+    await fetch('GET', 'http://localhost:8888/report_result?0');
```
Comment (dschuff) [inline test/browser/cwrap_early.js:12]: Does this need the 'GET'?

File: test/browser/test_offset_converter.c [hunk 3]
```diff
@@ -6,9 +6,7 @@ void *get_pc(void) { return __builtin_return_address(0); }
 int magic_test_function(void) {
   int result = EM_ASM_INT({
     function report(x) {
-      var xhr = new XMLHttpRequest();
-      xhr.open('GET', encodeURI('http://localhost:8888?stdout=' + x));
-      xhr.send();
+      fetch(encodeURI('http://localhost:8888?stdout=' + x));
```
Comment (dschuff) [inline test/browser/test_offset_converter.c:9]: ditto

File: test/browser_reporting.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ var hasModule = typeof Module === 'object' && Module;
 
 /** @param {boolean=} sync
     @param {number=} port */
-function reportResultToServer(result, sync, port) {
+function reportResultToServer(result, port) {
```
[the remark this answers is not in the document]
Comment (sbc100) [reply test/browser_reporting.js:5]: Luckily yes, not used.   Should I split out that change maybe?

[the remark this answers is not in the document]
Comment (dschuff) [reply test/browser_reporting.js:5]: I could go either way. If you split it out you'll just have to change this same code twice, and you've already got this PR so up to you.

File: test/test_browser.py [hunk 5]
```diff
@@ -4301,7 +4294,7 @@ def test_utf16_textdecoder(self):
     self.btest_exit('benchmark/benchmark_utf16.cpp', 0, args=['--embed-file', test_file('utf16_corpus.txt') + '@/utf16_corpus.txt', '-sEXPORTED_RUNTIME_METHODS=UTF16ToString,stringToUTF16,lengthBytesUTF16'])
 
   @parameterized({
-    '': ([],),
+    '': (['-g'],),
```
Comment (kripken) [inline test/test_browser.py:4297]: unintentional?

**Review by kripken — APPROVED** [review]

File: test/test_browser.py (hunk 5 above)
Comment (sbc100) [reply test/test_browser.py:4297]: Reverted.

**Pull request merged**